alpha/apps/kaltura/lib/kEntryPlayingDataHelper.php

```diff
@@ -0,0 +1,443 @@
+
+				if (count($deliveryProfileFlavorsForDc))
+				{
+					foreach ($deliveryProfileFlavorsForDc as $flavorAssetForDc)
+						$dcFlavorIds[] = $flavorAssetForDc->getId();
+					$sources[] = new KalturaPlayingSource($deliveryProfile->getId(), $deliveryProfile->getStreamerType(), $deliveryProfile->getPriority(), $this->constructProtocol($deliveryProfile), $dcFlavorIds, null, $flavorToDrmData);
+				}
+			}
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * @param $flavorAssets
+	 * @param $flavorAssetsIds
+	 * @param $flavorAssetsParamsNotIn
+	 */
+	protected function filterFlavorAssets(&$flavorAssets, $flavorAssetsIds, $flavorAssetsParamsNotIn)
+	{
+		foreach ($flavorAssets as $key => $flavorAsset)
+		{
+			if (in_array($flavorAsset->getId(), $flavorAssetsIds))
+			{
+				if ($flavorAssetsParamsNotIn)
+					unset($flavorAssets[$key]);
+			} else
+			{
+				if (!$flavorAssetsParamsNotIn)
+					unset($flavorAssets[$key]);
+			}
+		}
+	}
+
+	/**
+	 * @param $deliveryProfiles
+	 * @param $deliveryProfileIds
+	 * @param $deliveryProfilesParamsNotIn
+	 */
+	protected function filterDeliveryProfiles(&$deliveryProfiles, $deliveryProfileIds, $deliveryProfilesParamsNotIn)
+	{
+		foreach ($deliveryProfiles as $key => $deliveryProfile)
+		{
+			if (in_array($deliveryProfile->getId(), $deliveryProfileIds))
+			{
+				if ($deliveryProfilesParamsNotIn)
+				{
+					unset($deliveryProfiles[$key]);
+				}
+			} else
+			{
+				if (!$deliveryProfilesParamsNotIn)
+					unset($deliveryProfiles[$key]);
+			}
+		}
+	}
+
+	protected function getStreamsTypeToExclude($localCustomDeliveries)
+	{
+		$streamTypes = array();
+		foreach ($localCustomDeliveries as $deliveryProfile)
+			$streamTypes[] = $deliveryProfile->getStreamerType();
+
+		return $streamTypes;
+	}
+
+	/* @param $dbEntry
+	 * @param $flavorAssets
+	 * @param $deliveryProfile
+	 * @return KalturaEntryPlayingDataResult
+	 */
+	protected function getDrmData($dbEntry, &$flavorAssets, $deliveryProfile)
+	{
+		$playingDataParams = new KalturaEntryPlayingDataParams();
+		$playingDataParams->deliverProfile = $deliveryProfile;
+		$playingDataParams->flavors = KalturaFlavorAssetArray::fromDbArray(array_values($flavorAssets));
+
+		$result = new KalturaEntryPlayingDataResult();
+		$result->pluginData = new KalturaPluginDataArray();
+
+		$pluginInstances = KalturaPluginManager::getPluginInstances('IKalturaEntryPlayingDataContributor');
+		foreach ($pluginInstances as $pluginInstance)
+		{
+			$pluginInstance->contributeToEntryPlayingDataResult($dbEntry, $playingDataParams, $result);
+		}
+
+		if (count($result->flavorIdsToRemove))
+			$this->filterFlavorAssets($flavorAssets, $result->flavorIdsToRemove, true);
+
+		return $result->pluginData;
+	}
+
+
+	public function sortSources($localSources, $remoteSources, $servePriority)
+	{
+		$this->sortSourcesByPriority($localSources);
+		$this->sortSourcesByPriority($remoteSources);
+
+		switch ($servePriority)
+		{
+			case KalturaStorageServePriority::KALTURA_ONLY:
```

Comment:
Use core StorageProfile::STORAGE_SERVE_PRIORITY_KALTURA_ONLY instead, same for other enums.